### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by povilasb**
And update `travis` config to fail if `clippy`produces any warnings.

### CODE REVIEW

File: examples/udp_bind_public.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//! Demonstrates how bind to public address (using UPnP) and receive some data
+//! Demonstrates how to bind to public address (using `UPnP`) and receive some data
```
Comment (canndrew) [inline examples/udp_bind_public.rs]: Shame it demands "`" around UPnP. Wasn't there some way to disable this or add UPnP to its list of allowed words?

Comment (povilasb) [reply examples/udp_bind_public.rs]: There is: https://github.com/maidsafe/crust/blob/dev/clippy.toml
I just figured that if markdown was rendered, those ticks would actually make sense.
What do you think?

Comment (canndrew) [reply examples/udp_bind_public.rs]: How so? We don't have a type or trait in our code named `UPnP`.

Comment (canndrew) [reply examples/udp_bind_public.rs]: I'm just going to approve this though since it's not worth blocking this PR over something so trivial.

**Review by canndrew — APPROVED** [review]

**Review by canndrew — APPROVED** [review]

**Review by ustulation — APPROVED** [review]

**Pull request merged**